modules/openapi-generator/src/test/java/org/openapitools/codegen/markdown/MarkdownSampleGeneratorTest.java

```diff
@@ -1,3 +1,4 @@
+/*
 package org.openapitools.codegen.markdown;
```

Comment:
This is not part of my work, I only commented this test file because it caused errors during "mvm clean install"... The tests were crashing and preventing the build to complete.

Reply:
What JDK version, OS are you using? 

I saw 1 or 2 reports of similar issue before but couldn't repeat it. CIs (travis, circleci, github actions etc) are not able to repeat the issue either.

If you add `-DskipTests`, will you still get the same errors?

Reply:
Like I said in private in Slack, with -DskipTests, there are no errors.

Without `-DskipTests`, here is the errors we get. We are using Windows 10 and using either jdk 1.8 or openjdk 11.

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for openapi-generator-project 5.3.0-SNAPSHOT:
[INFO]
[INFO] openapi-generator-project .......................... SUCCESS [  5.500 s]
[INFO] openapi-generator-core ............................. SUCCESS [ 17.110 s]
[INFO] openapi-generator (core library) ................... FAILURE [02:17 min]
[INFO] openapi-generator (executable) ..................... SKIPPED
[INFO] openapi-generator (maven-plugin) ................... SKIPPED
[INFO] openapi-generator-gradle-plugin (maven wrapper) .... SKIPPED
[INFO] openapi-generator-online ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:40 min
[INFO] Finished at: 2021-09-23T13:32:16-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project openapi-generator: There are test failures.
[ERROR]
[ERROR] Please refer to C:\dev\Tools\openapi-generator\modules\openapi-generator\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C ""C:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot\bin\java" -javaagent:C:\\Users\\jcote\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.5\\org.jacoco.agent-0.8.5-runtime.jar=destfile=C:\\de
v\\Tools\\openapi-generator\\modules\\openapi-generator\\target\\jacoco.exec,excludes=**/gradle-wrapper.jar -XX:+StartAttachListener org.apache.maven.surefire.booter.ForkedBooter C:\Users\jcote\AppData\Local\Temp\surefire11650442259
439979018 2021-09-23T13-30-34_406-jvmRun1 surefire3748020510651876417tmp surefire_09914373202848665463tmp"
[ERROR] Process Exit Code: 0
[ERROR] Crashed tests:
[ERROR] org.openapitools.codegen.java.JavaClientCodegenTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C ""C:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot\bin\java" -javaagent:C:\\Users\\jcote\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.5\\org.jacoco.agent-0.8.5-runtime.jar=destfile=C:\\de
v\\Tools\\openapi-generator\\modules\\openapi-generator\\target\\jacoco.exec,excludes=**/gradle-wrapper.jar -XX:+StartAttachListener org.apache.maven.surefire.booter.ForkedBooter C:\Users\jcote\AppData\Local\Temp\surefire11650442259
439979018 2021-09-23T13-30-34_406-jvmRun1 surefire3748020510651876417tmp surefire_09914373202848665463tmp"
[ERROR] Process Exit Code: 0
[ERROR] Crashed tests:
[ERROR] org.openapitools.codegen.java.JavaClientCodegenTest
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:670)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:283)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1161)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1002)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:848)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
[ERROR]         at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
[ERROR]         at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:61)
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :openapi-generator
```

Reply:
Hello @wing328 ,
Like @JFCote mentionned we both get those kind errors. I am using jdk 1.8. For me it is for those two test files: 

- ProtobufSchemaCodegenTest.java
- MarkdownSampleGeneratorTest.java

If you want, I can uncomment back those two files and the "test check" for the pull request might fail. But since theses issues are clearly not caused by my changes, maybe you are willing to accept the merge anyway. Should I do that?

Reply:
I just pushed a fix to another branch [surefire-fix](https://github.com/OpenAPITools/openapi-generator/tree/surefire-fix).

Can you please give it a try to see if the fix works for your setup?

Reply:
ok, I just tried merging your surefix-fix branch it and unfortunately it does not fix the issue... I will try pushing anyway with my two files uncommented and see if it passes the checks in github...

Reply:
@wing328 All tests have succeeded, can you please proceed with the merge? (As indicated, I still have compilation issues with the two aforementioned files locally (my computer is windows 10 with java 1.8) but at least it seems to work in github...

Reply:
I've pushed more fixes to the branch [surefire-fix](https://github.com/OpenAPITools/openapi-generator/tree/surefire-fix).

Can you please give it a try without merging it into your branch?

Reply:
@wing328 I tried it directly. I checkout the `surefire-fix` branch and try this:
`./mvnw clean install` and I get this error:

```
[ERROR] Failures: 
[ERROR]   MarkdownSampleGeneratorTest.testSampleMarkdownGeneration:47 expected [# Object
## Properties

Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**Number** | **BigDecimal** |  | [optional] [default to null]
**Integer** | **Integer** |  | [optional] [default to null]
**Boolean** | **Boolean** |  | [optional] [default to null]
**String** | **String** |  | [optional] [default to null]
**CharFormat** | **String** |  | [optional] [default to null]
**DateFormat** | **date** |  | [optional] [default to null]
**DateTimeFormat** | **Date** |  | [optional] [default to null]
**ByteFormat** | **byte[]** |  | [optional] [default to null]
**BinaryFormat** | **File** |  | [optional] [default to null]
**EmailFormat** | **String** |  | [optional] [default to null]
**UuidFormat** | **UUID** |  | [optional] [default to null]
**PasswordFormat** | **String** |  | [optional] [default to null]
**IntegerFormat** | **Integer** |  | [optional] [default to null]
**IntFormat** | **Integer** |  | [optional] [default to null]
**Int32Format** | **String** |  | [optional] [default to null]
**Int64Format** | **String** |  | [optional] [default to null]
**ShortFormat** | **Short** |  | [optional] [default to null]
**LongFormat** | **Long** |  | [optional] [default to null]
**DecimalFormat** | **BigDecimal** |  | [optional] [default to null]
**FloatFormat** | **Float** |  | [optional] [default to null]
**DoubleFormat** | **Double** |  | [optional] [default to null]
**HostnameFormat** | **String** |  | [optional] [default to null]
**Ipv4Format** | **String** |  | [optional] [default to null]
**Ipv6Format** | **String** |  | [optional] [default to null]
**UriFormat** | **URI** |  | [optional] [default to null]
**Map** | **Map** |  | [optional] [default to null]
**InlinedObject** | [**Object_InlinedObject**](Object_InlinedObject.md) |  | [optional] [default to null]
**ReferencedObject** | [**ReferencedObject**](ReferencedObject.md) |  | [optional] [default to null]
**ArrayOfStrings** | **List** |  | [optional] [default to null]
**StringObject** | [**String**](String.md) |  | [optional] [default to null]

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

] but found [# Object
## Properties

Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**Number** | **BigDecimal** |  | [optional] [default to null]
**Integer** | **Integer** |  | [optional] [default to null]
**Boolean** | **Boolean** |  | [optional] [default to null]
**String** | **String** |  | [optional] [default to null]
**CharFormat** | **String** |  | [optional] [default to null]
**DateFormat** | **date** |  | [optional] [default to null]
**DateTimeFormat** | **Date** |  | [optional] [default to null]
**ByteFormat** | **byte[]** |  | [optional] [default to null]
**BinaryFormat** | **File** |  | [optional] [default to null]
**EmailFormat** | **String** |  | [optional] [default to null]
**UuidFormat** | **UUID** |  | [optional] [default to null]
**PasswordFormat** | **String** |  | [optional] [default to null]
**IntegerFormat** | **Integer** |  | [optional] [default to null]
**IntFormat** | **Integer** |  | [optional] [default to null]
**Int32Format** | **String** |  | [optional] [default to null]
**Int64Format** | **String** |  | [optional] [default to null]
**ShortFormat** | **Short** |  | [optional] [default to null]
**LongFormat** | **Long** |  | [optional] [default to null]
**DecimalFormat** | **BigDecimal** |  | [optional] [default to null]
**FloatFormat** | **Float** |  | [optional] [default to null]
**DoubleFormat** | **Double** |  | [optional] [default to null]
**HostnameFormat** | **String** |  | [optional] [default to null]
**Ipv4Format** | **String** |  | [optional] [default to null]
**Ipv6Format** | **String** |  | [optional] [default to null]
**UriFormat** | **URI** |  | [optional] [default to null]
**Map** | **Map** |  | [optional] [default to null]
**InlinedObject** | [**Object_InlinedObject**](Object_InlinedObject.md) |  | [optional] [default to null]
**ReferencedObject** | [**ReferencedObject**](ReferencedObject.md) |  | [optional] [default to null]
**ArrayOfStrings** | **List** |  | [optional] [default to null]
**StringObject** | [**String**](String.md) |  | [optional] [default to null]

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

]
[ERROR]   ProtobufSchemaCodegenTest.testCodeGenWithAllOf:66->assertFileEquals:75 expected [/*
  OAI Specification example for Polymorphism

  No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package ;

import public "models/lizard_all_of.proto";
import public "models/snake_all_of.proto";
import public "models/cat_all_of.proto";
import public "models/dog_all_of.proto";

message Pet {

  string petType = 140636936;

  bool hasLegs = 159828448;

  bool lovesRocks = 499337491;

  string name = 3373707;

  string bark = 3016376;

}
] but found [/*
  OAI Specification example for Polymorphism

  No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package ;

import public "models/lizard_all_of.proto";
import public "models/snake_all_of.proto";
import public "models/cat_all_of.proto";
import public "models/dog_all_of.proto";

message Pet {

  string petType = 140636936;

  bool hasLegs = 159828448;

  bool lovesRocks = 499337491;

  string name = 3373707;

  string bark = 3016376;

}
]
[INFO]
[ERROR] Tests run: 1542, Failures: 2, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for openapi-generator-project 5.3.0-SNAPSHOT:
[INFO]
[INFO] openapi-generator-project .......................... SUCCESS [  3.872 s]
[INFO] openapi-generator-core ............................. SUCCESS [ 18.781 s]
[INFO] openapi-generator (core library) ................... FAILURE [02:22 min]
[INFO] openapi-generator (executable) ..................... SKIPPED
[INFO] openapi-generator (maven-plugin) ................... SKIPPED
[INFO] openapi-generator-gradle-plugin (maven wrapper) .... SKIPPED
[INFO] openapi-generator-online ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:46 min
[INFO] Finished at: 2021-09-28T08:25:53-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project openapi-generator: There are test failures.
[ERROR]
[ERROR] Please refer to C:\dev\Tools\openapi-generator\modules\openapi-generator\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :openapi-generator
```

Reply:
I reviewed the failed tests but looks but couldn't find any clue why the tests were failing as the expected and actual results look the same to me. Could this be related to encoding or line break setting in your local environment? 

Do you have another machine that you can give it another try?

Reply:
@wing328 We only have Windows 10 machine. I would suggest that since this issue is occuring in the master that if everything is ok with this merge request, we merge it. We have another PR in preparation building on this one and it would help to merge this one before we propose the second one. We will continue to help you with this bug in the tests after the merge of course,  just talk to me on slack.

Reply:
> Could this be related to encoding or line break

This is exactly what I also suspected. I also cannot find any difference between the two generated files, and yet it seems to trigger a mismatch error in the test. I am not aware of any special configuration on my window 10 machine regarding carriage return or line breaks (I only know that on unix they are not the same, but we all know that). Also since jcote is having the same issues, it is probably not a local configuration only on my computer... Do you have a windows 10 machine as well ?
In any case, the build has passed all checks on github and theses issues are not related to this pull request. I was hoping we could just merge it as-is... ?

Reply:
I've pushed more fixes to the `surefire-fix` branch. Can you guys please give it another try?

Reply:
@wing328 I just test the `surefire-fix` and it now compile perfectly!

Reply:
Thanks for the confirmation. I've merged the [fix](https://github.com/OpenAPITools/openapi-generator/pull/10529) into master.

@bflamand can you please also give it a try and if it works for you can you please merge the latest master into this branch and uncomment those test cases?

Reply:
@wing328 The latest master is now merged and everything works for me now.